snuba/utils/streams/kafka.py

```diff
@@ -45,35 +67,70 @@ class KafkaConsumer(Consumer[TopicPartition, int, bytes]):
     )
 
     def __init__(self, configuration: Mapping[str, Any]) -> None:
-        self.__consumer = ConfluentConsumer(configuration)
+        auto_offset_reset = configuration.get("auto.offset.reset", "largest")
+        if auto_offset_reset in {"smallest", "earliest", "beginning"}:
+            self.__resolve_partition_starting_offset = self.__resolve_partition_offset_earliest
+        elif auto_offset_reset in {"largest", "latest", "end"}:
+            self.__resolve_partition_starting_offset = self.__resolve_partition_offset_latest
+        elif auto_offset_reset == "error":
+            self.__resolve_partition_starting_offset = self.__resolve_partition_offset_error
+        else:
+            raise ValueError("invalid value for 'auto.offset.reset' configuration")
+
+        self.__consumer = ConfluentConsumer(
+            {**configuration, "auto.offset.reset": "error"}
```

Comment:
Wouldn't this make the consumer always fail if there is no offset committed on the given consumer group ? For example, what happens the first time we subscribe to the topic with a new consumer group ?

Reply:
The assignment callback handles the cold start case, but it'd probably be helpful for this to be noted here, and started more clearly in the docstring: "offsets are only reset during the assignment callback" is technically true, but assumes the reader makes the connection, which seems like a bit of a leap.

Reply:
Rewrote some of the documentation with a92c79f734a75d513c649ada51687b64302c55d1, hopefully this is clearer now.